server/bin/pbench-sync-satellite.sh

```diff
@@ -250,10 +253,26 @@ for host in $hosts ;do
         for x in $state_list; do
             echo "$host/TO-SYNC/$x" >> ${state_change_log}
         done
+        rm $logdir/$host/ok-checks.log
+    fi
+
+    if [[ ! -s $logdir/$host/fail-checks.log ]] ; then
+        rm $logdir/$host/fail-checks.log
+    fi
+
+    if [[ -z $(grep -q -E -v 'OK|FAILED' $logdir/$host/md5-checks.log) ]] ; then
```

Comment:
Please reread my comment about how to simplify this: adding -q is *NOT* the only thing you have to do; in fact, since it eliminates the output, the -z test will *always* succeed: that is wrong.